docs/integration_fetching_data.md

```diff
@@ -0,0 +1,146 @@
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    return coordinator
+
+
+class MyEntity(entity.Entity):
+
+    def __init__(self, coordinator):
+        self.coordinator = coordinator
+
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        return not self.coordinator.failed_last_update
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.coordinator.async_add_listener(
+            self.async_write_ha_state
+        )
+
+    async def async_will_remove_from_hass(self):
+        """When entity will be removed from hass."""
+        self.coordinator.async_remove_listener(
+            self.async_write_ha_state
+        )
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the light on.
+
+        Example method how to request data updates.
+        """
+        # Do the turning on.
+        # ...
+
+        # Update the data
+        await self.coordinator.async_request_refresh()
+
+    async def async_update(self):
+        """Update the entity.
+
+        Only used by the generic entity update service.
+        """
+        await self.coordinator.async_request_refresh()
+```
+
+## One API endpoint per device
+
+If your one API endpoint is mapping to multiple entities, please see the previous section. Even if there are many endpoints, it's best to use a coordinator per endpoint to manage this data.
+
+If you can map exactly one endpoint to a single device, you can fetch the data for this entity inside the `update()`/`async_update()` methods. Make sure polling is set to `True` and Home Assistant will call this method regularly.
+
+If your entities need to fetch data before being written to Home Assistan for the first time, pass `True` to the `add_entities` method: `add_entities([MyEntity()], True)`.
+
+You can control the polling interval for your integration by defining a `SCAN_INTERVAL` constant in your platform.
+
+```python
+from datetime import timedelta
+
+SCAN_INTERVAL = timedelta(seconds=5)
+```
+
+# Request Parallelism
+
+> This is an advanced topic.
+
+Home Assistant has built-in logic to make sure that integrations don't hammer services/APIs and consume all available resources in Home Assistant. This logic is built around limiting the number of parallel requests. This logic is automatically used during service calls and entity updates.
+
+Home Assistant controls the number of parallel requests by maintaining a [semaphore](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Semaphore). For example, if the semaphore allows 1 parallel connection, updates and service calls will wait if one is in progress. If the value is 0, the integration is itself responsible for limiting the number of parallel requests if necessary.
```

Comment:
This is a bit confusing. From the source code it looks like most of the times this is used like a boolean instead of a limiter. Am I understanding it correctly, that an integration could define `PARALLEL_UPDATES=10` to allow ten concurrent updates? Is this capped per entity? Platform? Integration?

```suggestion
Home Assistant controls the number of parallel updates (calls to `update()`) by maintaining a [semaphore](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Semaphore) per integration. For example, if the semaphore allows 1 parallel connection, updates and service calls will wait if one is in progress. If the value is 0, the integration is itself responsible for limiting the number of parallel requests if necessary.
```